liiatools/datasets/shared_functions/common.py

```diff
@@ -80,6 +80,25 @@ def inherit_property(stream, prop_name):
         yield event
 
 
+def save_year_error(input, la_log_dir):
+    """
+    Save errors to a text file in the LA log directory
+
+    :param input: The input file location, including file name and suffix, and be usable by a Path function
+    :param la_log_dir: Path to the local authority's log folder
+    :return: Text file containing the error information
+    """
+    filename = str(Path(input).resolve().stem)
```

Comment:
Any reason for the str() function?  the stem function should return a string if given one.  In other words, are you uncertain that "input" is a string?  If so, then maybe there should be tests against it early in the function.

Reply:
I wasn't sure if .stem would return a str(), can be removed if that's the case. "input" should always be a string and is stated in the documentation I'm pretty sure

Reply:
Yes, according to the function itself, it merely returns a substring or the original string.

```    
def stem(self):
        """The final path component, minus its last suffix."""
        name = self.name
        i = name.rfind('.')
        if 0 < i < len(name) - 1:
            return name[:i]
        else:
            return name
```

So, yes, I think you should be able to safely remove the str() as long as you do checks on input at the start (which I beleve we do)